src/io/utils.rs

```diff
@@ -10,35 +10,31 @@ use std::marker::PhantomData;
 
 use super::params::ObfuscationParams;
 
-const TAG_R_0: &[u8] = b"R_0";
-const TAG_R_1: &[u8] = b"R_1";
 const TAG_A_RLWE_BAR: &[u8] = b"A_RLWE_BAR";
-const _TAG_BGG_PUBKEY_FHEKEY_PREFIX: &[u8] = b"BGG_PUBKEY_FHEKY:";
 const TAG_A_PRF: &[u8] = b"A_PRF:";
 pub const TAG_BGG_PUBKEY_INPUT_PREFIX: &[u8] = b"BGG_PUBKEY_INPUT:";
 
-pub fn sample_public_key_by_idx<K: AsRef<[u8]>, S>(
+pub fn sample_public_key_by_id<K: AsRef<[u8]>, S>(
```

Comment:
Why rename?

Reply:
Because we no longer have `idx` but `level`. So I think `id` indicates a more generic notion of identifier